### PR TITLE
Collection mobile

### LINKED ISSUE #5076: Redesign Collection Mobile
task details in
* #5000 
<hr/>

image

* hero items still in the banner container
* socials are collapsed to max 3 icons
* under banner is description and stats
* collapsed control section - almost like on explore
* smaller cards - same as on explore (implemented in #4893)
* burger menu will work like this (implemented in #4893)

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #5076 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
image

### CODE REVIEW

**Comment by daiagi** [comment]
will require merge resolution with #5146 
need to merge #5120  before merging this

**Comment by exezbcz** [comment]
@daiagi yup, that is correct, there is no grid button on mobile - mobile nft cards are also bit different
image
I will have a look 👍

**Comment by exezbcz** [comment]
oki, menu icon into a button - like on mobile fixes, stroke + dropshadow
image

+ change the nft cards on mobile to the minimized version like on explore
image

**Comment by daiagi** [comment]
[] count icon on filter menu toggle button
Like sort have

**Comment by daiagi** [comment]
:heavy_check_mark:  oki, menu icon into a button - like on mobile fixes, stroke + dropshadow
:heavy_check_mark: change the nft cards on mobile to the minimized version like on explore
:heavy_check_mark: and with the animation, there is a like a flash in darkmode before the screen moves in in (solid bg with lowered opacity - white), not sure if this was applied to desktop as well but the icons in menu are delayed and then jump over the whole screen
:heavy_check_mark: count icon on filter menu toggle button (mobile only)
:heavy_check_mark: sort button after chain selector on explore (and collection) in mobile

**Comment by exezbcz** [comment]
niceee, its looking good

- please change the menu so it looks same as on explore 
  - changer are - "filter" to bold, remove dropshadow around reset all button
  
image

- when is sorting applied and filters not, the menu icon is not aligned with content around 
image
- looks like its broken on bigger touch screens as well
image

animation looks good now 👍 

Please remove the divider that was separating the breadcrumbs and content on desktop
image

And then we may delete buy now by default from collections? Thinking out loud. Wdyt @roiLeo

**Comment by daiagi** [comment]
> when is sorting applied and filters not, the menu icon is not aligned with content around

yep, notice it too, fixed already

**Comment by exezbcz** [comment]
the item count now is on weird spot due to removing the divider on all touch devices I think.
image
- also on desktop the divider is missing now

image


+ the creator name changes to black bold on mobile, it should remain blue normal weight
image

**Comment by daiagi** [comment]
It's due to cases where the identity pop over is present and cases where it's renderd as simple link

Is the blue color app wide? Can I change the default color of the link to be blue?

Thank you

**Comment by exezbcz** [comment]
@daiagi ah, oki

the link isnt blue everywhere, see gallery item detail - creator and owner are bold black

**Comment by daiagi** [comment]
> the item count now is on weird spot due to removing the divider on all touch devices I think.

apologies, it slipped my vision. fixed

>also on desktop the divider is missing now

sure? i could not reproduce it

* Firefox
image


* Brave
image

> the creator name changes to black bold on mobile, it should remain blue normal weight

Fixed

image

**Comment by roiLeo** [comment]
> And then we may delete buy now by default from collections? Thinking out loud. Wdyt @roiLeo

Sure, we should show all items on collection item page by default.

File: layouts/explore-layout.vue [hunk 1]
```diff
@@ -15,14 +19,14 @@
...
-        <section class="section">
+        <section class="section py-2">
           <div class="container">
             <h1 v-if="isExplore" class="title">{{ $t('explore') }}</h1>
 
             <ExploreTabsFilterSort />
           </div>
         </section>
-        <hr class="m-0" />
+        <hr class="text-color" />
```
Comment (roiLeo) [inline layouts/explore-layout.vue:29]: Now we have a whitespace if you remove this

Screenshot 2023-03-06 at 09-13-40 Denver waifus 🤎 KodaDot - NFT Market Explorer

Comment (daiagi) [reply layouts/explore-layout.vue:29]: yep yep, missed that.
Thank you

**Comment by exezbcz** [comment]
yes, that was wrong on my end then :D 

last changes:
- the whitespace 
- remove buy now by default from collection 

thanks!

**Comment by daiagi** [comment]
> yes, that was wrong on my end then :D
> 
> last changes:
> 
>     * the whitespace
> 
>     * remove buy now by default from collection
> 
> 
> thanks!

Done

**Review by roiLeo — COMMENTED** [review]
Don't know if it's on mobile spec but I see stuff changed:
- creator link should be blue
Screenshot 2023-03-06 at 11-38-42 Ancient Civilizations KodaDot - NFT Market Explorer
- explore layout spacing issue (compare it with beta)
Screenshot 2023-03-06 at 11-40-27 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer
- weird animation on close sidebar filter

File: components/collection/CollectionInfo.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    class="is-flex is-justify-content-space-between mobile-flex-direction-column gap">
+    <div class="is-flex is-flex-direction-column is-flex-grow-1 max-width">
+      <div v-if="collectionInfo?.currentOwner" class="is-flex mb-2">
+        <div class="mr-2">{{ $t('activity.creator') }}</div>
+        <nuxt-link :to="`/${urlPrefix}/u/${address}`">
+          <IdentityIndex
+            ref="identity"
+            :address="address"
+            link-class="has-text-link" />
+        </nuxt-link>
```
Comment (roiLeo) [inline components/collection/CollectionInfo.vue:12]:
```suggestion
        <nuxt-link :to="`/${urlPrefix}/u/${address}`" class="has-text-link">
          <IdentityIndex
            ref="identity"
            :address="address" />
        </nuxt-link>
```

File: components/identity/module/IdentityChain.vue [hunk 3]
```diff
@@ -50,6 +51,7 @@ defineProps<{
   address?: Address
   shortenedAddress?: string
   name?: string | object
+  linkClass?: string
```
Comment (roiLeo) [inline components/identity/module/IdentityChain.vue:54]: I would remove this prop

**Review by roiLeo — COMMENTED** [review]
`collectionStatsByIdWithOffers` retrieve all data on client browser :/

File: components/collection/CollectionInfo.vue (hunk 2 above)
Comment (daiagi) [reply components/collection/CollectionInfo.vue:12]: I'm afraid this alone is not enough
on mobile, and maybe other cases, i did not explore deeply, ```IdenitityIndex``` renders ```IdentityLink``` and the applied class on ```nuxt-link``` just doesn't work

I've solved it before using css in ```collectionInfo``` which you requested changed

so this time i pass a prop through

on desktop, or any other case where ```IdentityIndex``` renders ```IdenitityPopover``` your solution is required

**Comment by daiagi** [comment]
> weird animation on close sidebar filter

exists on beta as well, should solve in separate issue

> explore layout spacing issue (compare it with beta)

indeed, apologies. fixed

**Comment by roiLeo** [comment]
> if i remove the `has-text-link` class from the internal `<a>` tag the text is black

Can't reproduce it, we have same stuff going in `GalleryItemDescription` component so I want to unify how it looks

File: components/collection/CollectionInfo.vue (hunk 2 above)
Comment (roiLeo) [reply components/collection/CollectionInfo.vue]: is it still a problem?

**Comment by daiagi** [comment]
> > if i remove the `has-text-link` class from the internal `<a>` tag the text is black
> 
> Can't reproduce it, we have same stuff going in `GalleryItemDescription` component so I want to unify how it looks

toggle mobile device in dev tools and make sure it us "mobile" mode aka  touch enabled. refresh page

image

from ```IdentityLink.vue```
image


maybe remove this ```!isMobileDevice```  => render identityPopover on touch devices as well ? wtyt @roiLeo

**Comment by roiLeo** [comment]
> toggle mobile device in dev tools and make sure it is in "mobile" mode aka touch enabled. refresh page

Aight, I can reproduce it with mobile mode.

> from `IdentityLink.vue` 

This is where issue come from, since `IdentityLink` provide a nested `<a/>` tag, IMO we should replace this component by something else since when you change `a` tag to `div` it work like a charm

> maybe remove this `!isMobileDevice` => render identityPopover on touch devices as well ? wtyt @roiLeo

yep easy solution, but I think this has been introduce for a particular reason

**Comment by daiagi** [comment]
Mind approving this PR and I'll open a new issue to deal with this?

**Review by roiLeo — APPROVED** [review]
What need to be improved:
- profile link on mobile (`IdentityLink` component)
- use `NeoSidebar` instead of Buefy Sidebar (`MobileFilter` component)
- `index` folder in `/components/explore` ???
- remove `linkClass` prop
- remove `class="mb-5"` (total in `ItemsGrid` & `Gallery` component)
- change `position: relative` with `is-relative` helper
- `v-if` on `ActiveCount` usage

otherwise lgtm

**Comment by daiagi** [comment]
> What need to be improved:
> - profile link on mobile (`IdentityLink` component)
> - use `NeoSidebar` instead of Buefy Sidebar (`MobileFilter` component)
> - `index` folder in `/components/explore` ???
> - remove `linkClass` prop
> - remove `class="mb-5"` (total in `ItemsGrid` & `Gallery` component)
> - change `position: relative` with `is-relative` helper
> - `v-if` on `ActiveCount` usage

All done except:

#### Leftovers

- [ ] profile link on mobile (IdentityLink component)
- [ ] remove linkClass prop

**Comment by daiagi** [comment]
@Jarsen136 @preschian  could i have your review please?

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Pull request merged**